Mathlib/Data/BitVec/Defs.lean

```diff
@@ -140,4 +140,14 @@ def toLEList (x : BitVec w) : List Bool :=
 def toBEList (x : BitVec w) : List Bool :=
   List.ofFn x.getMsb'
 
+/-- Create a bitvector from a function that maps index `i` to the `i`-th least significant bit -/
+def ofLEFn {w} (f : Fin w → Bool) : BitVec w :=
```

Comment:
Do you think it makes more sense to implement this by taking the bitwise or of `bif f i then 1 else 0 <<< i` using `multiset.fold` or similar?
If nothing else, it would be nice to prove that those are equal

Reply:
I considered this, it might be faster since there are then less data-dependencies, but it does complicate proofs.
In particular, we might want to upstream `ofLEFn` to Std at some point, so I would prefer not to rely to much on Mathlib-specific APIs.

For posterity: the following is an alternative def of `ofLEFn` that would work.
```lean
List.finRange w
    |>.map (fun i =>
        shiftLeftZeroExtend (ofBool (f i)) i.val |>.zeroExtend' (Nat.add_comm _ _ ▸ i.prop)
      )
    |>.foldr (· ||| ·) 0#_
```